### ISSUE TITLE
feat: adding abstraction of optimal agent movement

### ISSUE THREAD

**Issue by adamamer20** [projectmesa/mesa-frames#108]
It would be beneficial to have a function in the DiscreteSpaceDF (and AgentContainer) for moving agents based on attributes:

```py
move_to(agents: AgentLike, attr_names: str | list[str], rank_order: ['max' | 'min'] | list['max' | 'min'])
```

For example: `move_to('sugar', 'max')`. This would abstract the code for handling sequential conflicts, making it easier than handling them from scratch each time.

**Comment by suryanshgargbpgc** [comment]
hi @adamamer20, I tried to understand this issue, if I am correct this feature aims to write a function which upon calling would solve the issue of multiple agents trying to move on the same grid at the same time which will prevent us from dealing with those edge cases everytime. 
Can you please share me necessary resources which can help me address this issue ?
Also could you please elaborate on any points i missed.

**Comment by adamamer20** [comment]
@suryanshgargbpgc You're correct in your understanding! I recommend checking out the sugarscape example for polars to get a clear picture of these concurrency conditions. You can see it here:

https://github.com/projectmesa/mesa-frames/blob/1d7b3ed11a61130cbee12fe0c9222d233b234669/examples/sugarscape_ig/ss_polars/agents.py#L174C1-L225C73

In this example, the best move might be the same for multiple agents at the same time. Since we're moving all ants simultaneously in a vectorized way, it's important to address this issue. I handled it this way in sugarscape, and I believe the same reasoning can be generalized to accommodate various agent attribute objectives, removing the burden of the implementation on the user

**Comment by suryanshgargbpgc** [comment]
Hi @adamamer20, sorry I had my exams, I was thinking the function could look something like 

First, the function gathers all possible moves for each agent into a list. Each move records the candidate grids’ attribute value (like sugar) and a random number when conflict occurs. Next, the list is sorted so that grids with the best sugar value comes first (with the random number resolving ties). Then, the function iterates through the sorted moves and assigns each agent to the target grid if that agent hasn’t been assigned a move yet and the grid is still free. If an agent doesn’t secure a new grid, it remains in its original position. Finally, the agents’ positions are updated accordingly.
We can iterate through the list for low to high attribute value if the agent is avoiding the grid.

does it look fine?
